File: Search/FindMinimumInRotatedSortedArray/FindMinimumInRotatedSortedArray.py
class Solution:
    def findMin(self, nums):
        if not nums:
            return -1
            
        start, end = 0, len(nums) - 1
        
        while start + 1 < end:
            mid = start + (end - start) / 2
            if nums[mid] < nums[end]:
                end = mid
            else:
                start = mid
                
        if nums[start] < nums[end]:
            return nums[start]
        else:
            return nums[end]

File: Search/FindMinimumInRotatedSortedArray/FindMinimumInRotatedSortedArray.cc
class Solution {
	public:
		int findMin(vector<int> &nums) {
			if(nums.empty()) {
				return -1;
			}
			
			int start = 0;
			int end = nums.size() - 1;
			int mid = 0;
			
			while(start +  1 < end) {
				mid = start + (end - start) / 2;
				if(nums[mid] < nums[end]) 
					end = mid;
				else 
					start = mid;
			}
			
			if(nums[start] < nums[end])
				return nums[start];
			else
				return nums[end];
		}
};